packages/toast/src/Toast.tsx

```diff
@@ -0,0 +1,301 @@
+      color: ${uiColors.gray.dark2};
+    `,
+
+    contentWrapper: css`
+      border-radius: 4px 4px 0 0;
+      border: 1px solid ${uiColors.blue.light2};
+      border-bottom: 0;
+    `,
+
+    body: css`
+      color: ${uiColors.gray.dark2};
+    `,
+  },
+};
+
+const RTGStates = {
+  Entering: 'entering',
+  Entered: 'entered',
+  Exiting: 'exiting',
+  Exited: 'exited',
+} as const;
+
+type RTGStates = typeof RTGStates[keyof typeof RTGStates];
+
+const toastTransitionStateStyles: Partial<Record<RTGStates, string>> = {
+  [RTGStates.Entered]: css`
+    transform: translate3d(0, 0, 0) scale(1);
+    opacity: 1;
+  `,
+};
+
+interface ToastProps extends Omit<React.ComponentProps<'div'>, 'title'> {
+  /**
+   * Optional text shown in bold above the body text.
+   */
+  title?: React.ReactNode;
+
+  /**
+   * Required main text for the Toast.
+   */
+  body: React.ReactNode;
+
+  /**
+   * Optional className passed to the wrapping <div /> for the toast.
+   */
+  className?: string;
+
+  /**
+   * Required style variant to render the Toast as.
+   */
+  variant: ToastVariants;
+
+  /**
+   * Optional number between 0 and 1 that sets the progress bar's progress. Note that the progress bar is only rendered when the Toast variant is set to `'progress'`.
+   *
+   * **Default:** `1`
+   */
+  progress?: number;
+
+  /**
+   * Optional boolean that renders the Toast and makes it visible when true.
+   *
+   * **Default:** `false`
+   */
+  open?: boolean;
+
+  /**
+   * Optional click event handler that, when set, renders a close button that receives the passed handler.
+   */
+  close?: React.MouseEventHandler;
+}
+
+function Toast({
+  title,
+  body,
+  className,
+  variant,
+  progress = 1,
+  open = false,
+  close,
+  ...rest
+}: ToastProps) {
+  const nodeRef = useRef(null);
+  const dismissible = !!close;
+
+  let VariantIcon: React.ComponentType<any>;
+
+  if (variant === ToastVariants.progress) {
+    VariantIcon = RefreshIcon;
+  } else if (variant === ToastVariants.success) {
+    VariantIcon = CheckmarkWithCircleIcon;
+  } else {
+    VariantIcon = CheckmarkWithCircleIcon;
+  }
+
+  const currentVariantStyles = variantStyles[variant];
+
+  return (
+    <Transition
+      in={open}
+      timeout={transitionDuration}
+      mountOnEnter
+      unmountOnExit
+      nodeRef={nodeRef}
+    >
+      {(state: RTGStates) => (
+        <Portal>
+          <div
+            role="status"
+            ref={nodeRef}
+            className={cx(
+              baseElementStyles.toast,
+              currentVariantStyles.toast,
+              toastTransitionStateStyles[state],
+              className,
+            )}
+            {...rest}
+          >
+            <div
+              className={cx(
+                baseElementStyles.contentWrapper,
+                currentVariantStyles.contentWrapper,
+                {
+                  [css`
+                    padding-right: ${spacing[3] * 2}px;
+                  `]: dismissible,
```

Comment:
Woah, funky.

Reply:
Gotta make sure that the text doesn't collide with the button or its state. It's possible that doing ``${spacing[3] * 2}px`` is actually misleading, and rather this should be ``${spacing[3] + ${magicNumber}}px``.

Reply:
Oh I meant more, inlining the CSS function in the key value. Never thought to do that, so caught me by surprise.

Reply:
Ahh gotcha 😄